### ISSUE TITLE
About the logging the warped image

### ISSUE THREAD

**Issue by gengrui1983** [ClementPinard/SfmLearner-Pytorch#47]
Hi Clement,

Thank you very much for your implementation. I just got a question on this line:

https://github.com/ClementPinard/SfmLearner-Pytorch/blob/e14f747288683a26406a13399614c19c1b118a00/train.py#L413

Maybe I think it incorrectly, but I found in the process of logging the warped image, the code seems that applies the pose t+1 on the ref_img t-1? Could you please kindly help to explain this line a bit? 

Thank you very much.

Best regards,
Rui

**Comment by ClementPinard** [comment]
hello, the `:1` here refers to the fact that for logging purpose only, we construct a fake batch of only one element, to make things faster. You could argue that it could have been done directly from the actual batch computing in the photometric loss function. It's actually what I did in this version of the code : https://github.com/ClementPinard/unsupervised-depthnet/blob/master/train_img_pairs.py#L381

I did not put it here because it needs a bit of refactoring (but it might come !)

as such everything behaves like the photometric loss function here : https://github.com/ClementPinard/SfmLearner-Pytorch/blob/master/loss_functions.py#L26

`ref_imgs` is a list of imgs tensors (from t-2 to t+2, excluding t), `pose` is a tensor of poses, hence the necessity of the enumerate in the line before.

But here, ref is the jth element in the list of ref_imgs, and thus it matches with the element `pose[:,j]`

**Comment by gengrui1983** [comment]
Thank you @ClementPinard !
